open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/manage/vrp/VRPConsentRequestValidatorTest.java

```diff
@@ -0,0 +1,2630 @@
+        JSONObject testData3 = new JSONObject();
+
+        JSONObject result3 = VRPConsentRequestValidator.validateJsonObjectKey(testData3, "currency");
+        //Assert.assertFalse(result3);
+
+        // Test case 4: Invalid currency key (null parentObj)
+        JSONObject result4 = VRPConsentRequestValidator.validateJsonObjectKey(null, "currency");
+        //Assert.assertFalse(result4);
+
+        // Add more test cases as needed
+    }
+
+
+    @Test
+    public void testValidationFailureForNullCurrencyKey() {
+
+        JSONArray periodicLimits = new JSONArray();
+        periodicLimits.add(new JSONObject());
+
+        JSONObject result = VRPConsentRequestValidator.validateAmountCurrencyPeriodicLimits(periodicLimits,
+                ConsentExtensionConstants.CURRENCY);
+
+        Assert.assertFalse((boolean) result.get(ConsentExtensionConstants.IS_VALID));
+        Assert.assertEquals("Mandatory parameter 'Currency' of periodic limits is not present in payload",
+                result.get(ConsentExtensionConstants.ERRORS));
+    }
+
+    @Test
+    public void testVrpInitiationPayloadWithoutPeriodicTypeCurrency() {
+
+        String initiationPayloads = VRPTestConstants.METADATA_VRP_WITHOUT_PERIODIC_TYPE_CURRENCY;
+        JSONObject result = VRPConsentRequestValidator.validateControlParameters((JSONObject) JSONValue.
+                parse(initiationPayloads));
+        JSONObject result2 = VRPConsentRequestValidator.validatePeriodicLimits((JSONObject) JSONValue.
+                parse(initiationPayloads));
+
+        Assert.assertFalse((boolean) result.get(ConsentExtensionConstants.IS_VALID));
+        Assert.assertFalse((boolean) result2.get(ConsentExtensionConstants.IS_VALID));
+        Assert.assertEquals(ErrorConstants.MISSING_MAXIMUM_INDIVIDUAL_AMOUNT,
+                result.get(ConsentExtensionConstants.ERRORS));
+    }
+
+
+    @Test
+    public void testValidationFailureForMissingKey() {
+
+        JSONArray periodicLimits = new JSONArray();
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("otherKey", "someValue");
+        periodicLimits.add(jsonObject);
+
+
+        JSONObject result = VRPConsentRequestValidator.validateAmountCurrencyPeriodicLimits(periodicLimits,
+                ConsentExtensionConstants.CURRENCY);
+        Assert.assertFalse((boolean) result.get(ConsentExtensionConstants.IS_VALID));
+        Assert.assertEquals("Mandatory parameter 'Currency' of periodic limits is not present in payload",
+                result.get(ConsentExtensionConstants.ERRORS));
+
+    }
+
+    @Test
+    public void testValidationFailureForNullParentArray() {
+
+        // Call the method with a null parentArray
+        JSONObject result = VRPConsentRequestValidator.validateAmountCurrencyPeriodicLimits(null,
+                ConsentExtensionConstants.CURRENCY);
+
+        // Assert that the result is false, indicating a validation failure for a null parentArray
+        //Assert.assertFalse(result);
```

Comment:
Can you re-check these commented assertions? (check above methods as well)

Reply:
Fixed in [876c68f]